tabular/src/autogluon/tabular/predictor/predictor.py

```diff
@@ -158,18 +165,22 @@ def __init__(
             eval_metric=None,
             path=None,
             verbosity=2,
+            sample_weight=None,
+            weight_evaluation=False,
             **kwargs
     ):
         self.verbosity = verbosity
         set_logger_verbosity(self.verbosity, logger=logger)
+        self.sample_weight = sample_weight  # TODO: add support for sample_weight='auto'
```

Comment:
also TODO: support 'balanced'

Reply:
I think we should do a follow-up PR to add 'balanced' support (and default 'auto' to 'balanced'), as it will be a nice option for users to try out without needing to understand how to construct a proper sample_weight column, and probably will be better than nothing. We can just construct based off of class frequency for binary/multiclass, and raise exception on regression.